uni/lib/view/home/home.dart

```diff
@@ -13,9 +15,33 @@ class HomePageView extends StatefulWidget {
 
 /// Tracks the state of Home page.
 class HomePageViewState extends GeneralPageViewState {
+  bool? isBannerViewed;
+
+  @override
+  void initState() {
+    super.initState();
+    checkBannerViewed();
+  }
+
+  Future<void> checkBannerViewed() async {
+    isBannerViewed = await AppSharedPreferences.isDataCollectionBannerViewed();
+    await AppSharedPreferences.setDataCollectionBannerViewed(isViewed: true);
+    setState(() {});
+  }
+
   @override
   Widget getBody(BuildContext context) {
-    return const MainCardsList();
+    return (isBannerViewed ?? false)
```

Comment:
- Considering `isBannerViewed` as false before finishing the get from the preferences means that the framework is free to render the banner for a few milliseconds always. I would make it non nullable and init it as true by default.
- Use a local variable to get from the preferences and assign it to the set inside setState.